pkg/app/run.go

```diff
@@ -622,7 +622,7 @@ func commandRequiresAuthServer(command string) bool {
 // requires an API token.
 func commandRequiresToken(command string) bool {
 	switch command {
-	case "compute build", "compute hash-files", "compute init", "compute metadata", "compute serve":
+	case "compute build", "compute hash-files", "compute metadata", "compute serve":
```

Comment:
This is an interesting scenario because by making `compute init` return `true` (i.e. "it requires a token") this potentially causes issues for users who want to mess around with the Fastly platform locally but who don't have an account or have a token created yet (e.g. they want to try out Fastly and to spin up some application code locally, i.e. `fastly compute serve` without needing to create an account).

I wonder if we could add additional logic to https://github.com/fastly/cli/blob/f3790ccee587fd63f3879f7e8a860091f16d4e29/pkg/app/run.go#L232 so that we check if the `--from` flag was passed and if it's not a Service ID, then we'll skip that conditional branch that tries to validate a token.

Reply:
Yeah, I'm going to take a look at what this would take. I also thought about what it would look like to defer this check until an authenticated API request needs to happen, to avoid needing to pre-list the potential commands. The simplest approach (check to see if `command` matches `/^compute init --from(=|\s)[a-zA-Z0-9]{22}$/` (ish, pending updates from [Patrick's comment](https://github.com/fastly/cli/pull/1187#discussion_r1582989336)) also seems brittle, but may be fine for now, at least?

Reply:
tl;dr we needed access to more information here than was available through the command name. This preserves the majority of the logic, switching on the name, but changes the input to take the full command interface.

Reply:
Yeah this is fine, and nice lateral thinking 👍🏻